core/types/types.cc

```diff
@@ -412,7 +412,8 @@ OrType::OrType(const TypePtr &left, const TypePtr &right) : left(move(left)), ri
 
 void TupleType::_sanityCheck(const GlobalState &gs) const {
     sanityCheckProxyType(gs, underlying());
-    auto *applied = cast_type<AppliedType>(this->underlying());
+    auto underlying = this->underlying();
+    auto *applied = cast_type<AppliedType>(underlying);
```

Comment:
These two usages were exactly the problem (in TupleType). To confirm: the change you made caused these to become compile errors as written? You didn't hunt this call site down on your own?

Reply:
Yes, I made the relevant `= delete` changes and then all the other changes were required to fix the build.